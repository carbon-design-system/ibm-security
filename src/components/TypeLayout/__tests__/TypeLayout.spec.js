/**
 * @file Type layout tests.
 * @copyright IBM Security 2019
 */

import React from 'react';
import { shallow } from 'enzyme';

import { TypeLayout, TypeLayoutBody, TypeLayoutRow, TypeLayoutCell } from '../';

import rows from '../_mocks_';

describe('TypeLayout', () => {
  let typeLayout;

  beforeEach(() => {
    typeLayout = shallow(
      <TypeLayout>
        <TypeLayoutBody>
          {rows.map(row => {
            const { id, title, description } = row;

            return (
              <TypeLayoutRow key={id}>
                <TypeLayoutCell>{title}</TypeLayoutCell>
                <TypeLayoutCell>
                  <ul>
                    <li>{description}</li>
                    <li>{description}</li>
                    <li>{description}</li>
                  </ul>
                </TypeLayoutCell>
              </TypeLayoutRow>
            );
          })}
        </TypeLayoutBody>
      </TypeLayout>
    );
  });

  describe('render', () => {
    it('renders correctly', () => {
      expect(typeLayout).toMatchSnapshot();
    });

    it("renders the HTML of the node's subtree", () => {
      expect(typeLayout.render()).toMatchSnapshot();
    });

    it('renders correct classes if `size` or `border` are set', () => {
      typeLayout.setProps({ size: 'sm', border: true });
      expect(typeLayout.render()).toMatchSnapshot();
    });
  });
});
