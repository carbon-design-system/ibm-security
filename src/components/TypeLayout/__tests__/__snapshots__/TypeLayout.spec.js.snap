// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeLayout render renders correct classes if \`size\` or \`border\` are set 1`] = `
<section
  aria-label="Structured list section"
  class="bx--structured-list security--type-layout__container security--type-layout__container--border security--type-layout__container--sm"
>
  <div
    class="bx--structured-list-tbody security--type-layout__container__body"
  >
    <div
      class="bx--structured-list-row security--type-layout__container__row"
    >
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--sm bx--structured-list-td"
      >
        Name
      </div>
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--sm bx--structured-list-td"
      >
        <ul>
          <li>
            Some text entered here
          </li>
          <li>
            Some text entered here
          </li>
          <li>
            Some text entered here
          </li>
        </ul>
      </div>
    </div>
    <div
      class="bx--structured-list-row security--type-layout__container__row"
    >
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--sm bx--structured-list-td"
      >
        Longer Name
      </div>
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--sm bx--structured-list-td"
      >
        <ul>
          <li>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
          <li>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
          <li>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
        </ul>
      </div>
    </div>
    <div
      class="bx--structured-list-row security--type-layout__container__row"
    >
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--sm bx--structured-list-td"
      >
        Even Longer Name
      </div>
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--sm bx--structured-list-td"
      >
        <ul>
          <li>
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
          <li>
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
          <li>
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;

exports[`TypeLayout render renders correctly 1`] = `
<StructuredListWrapper
  ariaLabel="Structured list section"
  border={false}
  className="security--type-layout__container security--type-layout__container--md"
  selection={false}
>
  <TypeLayoutBody
    className=""
    key=".0"
  >
    <TypeLayoutRow
      className=""
      key=".$0"
    >
      <TypeLayoutCell
        className="security--type-layout__container__cell--md"
        key=".0"
      >
        Name
      </TypeLayoutCell>
      <TypeLayoutCell
        className="security--type-layout__container__cell--md"
        key=".1"
      >
        <ul
          key=".0"
        >
          <li
            key=".0"
          >
            Some text entered here
          </li>
          <li
            key=".1"
          >
            Some text entered here
          </li>
          <li
            key=".2"
          >
            Some text entered here
          </li>
        </ul>
      </TypeLayoutCell>
    </TypeLayoutRow>
    <TypeLayoutRow
      className=""
      key=".$1"
    >
      <TypeLayoutCell
        className="security--type-layout__container__cell--md"
        key=".0"
      >
        Longer Name
      </TypeLayoutCell>
      <TypeLayoutCell
        className="security--type-layout__container__cell--md"
        key=".1"
      >
        <ul
          key=".0"
        >
          <li
            key=".0"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
          <li
            key=".1"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
          <li
            key=".2"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
        </ul>
      </TypeLayoutCell>
    </TypeLayoutRow>
    <TypeLayoutRow
      className=""
      key=".$2"
    >
      <TypeLayoutCell
        className="security--type-layout__container__cell--md"
        key=".0"
      >
        Even Longer Name
      </TypeLayoutCell>
      <TypeLayoutCell
        className="security--type-layout__container__cell--md"
        key=".1"
      >
        <ul
          key=".0"
        >
          <li
            key=".0"
          >
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
          <li
            key=".1"
          >
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
          <li
            key=".2"
          >
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
        </ul>
      </TypeLayoutCell>
    </TypeLayoutRow>
  </TypeLayoutBody>
</StructuredListWrapper>
`;

exports[`TypeLayout render renders the HTML of the node's subtree 1`] = `
<section
  aria-label="Structured list section"
  class="bx--structured-list security--type-layout__container security--type-layout__container--md"
>
  <div
    class="bx--structured-list-tbody security--type-layout__container__body"
  >
    <div
      class="bx--structured-list-row security--type-layout__container__row"
    >
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--md bx--structured-list-td"
      >
        Name
      </div>
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--md bx--structured-list-td"
      >
        <ul>
          <li>
            Some text entered here
          </li>
          <li>
            Some text entered here
          </li>
          <li>
            Some text entered here
          </li>
        </ul>
      </div>
    </div>
    <div
      class="bx--structured-list-row security--type-layout__container__row"
    >
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--md bx--structured-list-td"
      >
        Longer Name
      </div>
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--md bx--structured-list-td"
      >
        <ul>
          <li>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
          <li>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
          <li>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. 
          </li>
        </ul>
      </div>
    </div>
    <div
      class="bx--structured-list-row security--type-layout__container__row"
    >
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--md bx--structured-list-td"
      >
        Even Longer Name
      </div>
      <div
        class="security--type-layout__container__cell security--type-layout__container__cell--md bx--structured-list-td"
      >
        <ul>
          <li>
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
          <li>
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
          <li>
            Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
